### DISCUSSION TITLE
Getting trim info in 3 different steps of the processing.

### DISCUSSION THREAD

**Discussion by bgflores** [ImageMagick/ImageMagick#7405] [Digital Image Processing]
I have the below command. The idea is to get the trim info of the original image after processing, then -chop left side, get the same info, then chop the left and get the same information.
Each part works great in isolation but if I join into 1 command line it becomes stuck before starting the first clone.
Any ideas?
magick -quiet "0008.tif" -blur 0x2 -threshold 50% -morphology open diamond:2 -morphology open square:1 -format "%@" info: /
( -clone 0 -gravity east -chop 55%x0  -format "%@" info: +delete ) /
( +clone -gravity west -chop  55%x0 -format "%@" info: +delete )

**Comment by bgflores** [comment]
I forgot
Windows 11 
 ImageMagick 7.1.0-51

**Comment by GeeMack** [comment]
ImageMagick expects the last argument in a command to be a filename, but you can write to files, or even to "info:" along the way. Put "-write info:" instead of just "info:" where you want the current specs to print out.

This command...

    magick logo: -trim -write info: -border 6x6 -write info: null:

... produces this result...

    logo: GIF 458x473 640x480+92+0 8-bit sRGB 256c 28576B 0.000u 0:00.000
    logo: GIF 470x485 652x492+92+0 8-bit sRGB 28576B 0.000u 0:00.000

Make sure to terminate the command with "info:", "null:", "-exit", or an output file name.

**Reply by bgflores** [reply]
Thanks. Great help. It seems to be working well. but looks like I only info on the last two interations.

Displayed information
"2186x3850+441+453write image[]: 5 of 6, 100% complete
2058x3796+0+516"

magick -quiet -monitor "0008.tif" -blur 0x2 -threshold 50% -morphology open diamond:2 -morphology open square:1 -write mpr:img0 +delete \ 
( mpr:img0 -format "%@" -write info: null: ) \ 
( mpr:img0 -gravity east -chop 55%x0 -format "%@" -write info: null: ) \ 
( mpr:img0 -gravity west -chop 55%x0 -format "%@" -write info: null: ) null:

**Reply by fmw42** [reply]
You have mixed Unix and Windows syntax.  Unix needs parentheses escaped and Windows does not. You can also just use +delete in place of null: inside the parentheses. Finally, you need to use \n new lines to make it more readable. So here I chop 25% of the barn image, so it starts at 400 wide and after each chop, it is 300 wide.

![barn](https://github.com/ImageMagick/ImageMagick/assets/13458218/19789f83-dd46-46ae-9791-625be8c58c67)


```
magick -quiet -monitor "barn.jpg" -blur 0x2 -threshold 50% -morphology open diamond:2 -morphology open square:1 -write mpr:img0 +delete \
\( mpr:img0 -format "\n%@\n" -write info: +delete \) \
\( mpr:img0 -gravity east -chop 25%x0 -format "\n%@\n" -write info: +delete \) \
\( mpr:img0 -gravity west -chop 25%x0 -format "%@\n" -write info: +delete \) \
null:
```

<br>

```
400x258+0+41
chop image[img0]: 298 of 299, 100% complete
chop image[img0]: 299 of 299, 100% complete
300x258+0+41
300x258+0+41

```

**Reply by bgflores** [reply]
Thanks for all the help and tips. It works great, I can iterate all my steps and get info on them that I use later in my process.
